extension/source/Controllers/QuillController.ts

```diff
@@ -512,28 +508,30 @@ export default class QuillController extends BaseController<
         return accounts;
       },
 
-      getAccounts: async () =>
+      eth_accounts: async () =>
         // Expose no accounts if this origin has not been approved, preventing
         // account-requiring RPC methods from completing successfully
         // only show address if account is unlocked
         this.selectedAddress ? [this.selectedAddress] : [],
 
-      getProviderState: async () => {
+      eth_coinbase: async () => this.selectedAddress || null,
+
+      wallet_get_provider_state: async () => {
         return {
           accounts: this.selectedAddress ? [this.selectedAddress] : [],
           chainId: this.networkController.state.chainId,
           isUnlocked: !!this.selectedAddress,
         };
       },
 
-      setPreferredAggregator: async (req: any) => {
+      eth_setPreferredAggregator: async (req: any) => {
```

Comment:
Similarly, could type these functions instead of `any`.

```suggestion
      eth_setPreferredAggregator: async (req: JRPCRequest<[string]>) => {
```

Reply:
Nah it's more complicated than that. The request data is ultimately an external source that could send us any json data so in order to add type information we need to add runtime type checking (e.g. using io-ts).

There's also `.tabId` which isn't in the type information for `JRPCRequest` so typescript thinks it's not there no matter what type we give to it.

Definitely in favour of improving our typing but all of this stuff is very half-hearted in its type information so it's difficult to improve that situation without doing a deeper dive.

Reply:
We could add `.tabId` by using an extended type (`TabIDJRPCRequest extends JRPCRequest...`) but I see your point about unknown external data. Would be good to sanitize/check it for anything malicious but since this is not intended to be a production wallet less of a priority.

Given that, in general I think we can switch many of our `any` types to `unknown`. This will force relevant type checks such as using `in` or `typeof` and explicit casts to the expected type. If we want to be really strict, we can enable https://www.typescriptlang.org/tsconfig#noImplicitAny & https://typescript-eslint.io/rules/no-explicit-any